File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        sol = []
        before_prod = [1] * len(nums)
        after_prod = [1] * len(nums)
        for i in range(1, len(nums)):
            before_prod[i] = before_prod[i-1] * nums[i-1]
        for i in range(len(nums) - 2, -1, -1):
            after_prod[i] = after_prod[i+1] * nums[i+1]
        for i in range(len(nums)):
            sol.append(before_prod[i] * after_prod[i])
        return sol

File: Data Structures & Algorithms/sort-colors/submission-0.java
class Solution {
    public void sortColors(int[] nums) {
        int[] track = new int[3];
        for(int i = 0; i < nums.length; i++){
            track[nums[i]]++;
        } 
        int j = 0;
        for(int k = 0; k < track.length; k++){
            while(track[k] > 0){
                nums[j] = k;
                j++;
                track[k]--;
            }
        }
    }
}